fsw/cfe-core/src/sb/cfe_sb_priv.c

```diff
@@ -736,99 +515,104 @@ void CFE_SB_FinishSendEvent(CFE_ES_ResourceID_t TaskId, uint32 Bit){
 **  Return:
 **
 */
-int32 CFE_SB_AddDest(CFE_SB_RouteEntry_t *RouteEntry, CFE_SB_DestinationD_t *NewNode){
+int32 CFE_SB_AddDestNode(CFE_SBR_RouteId_t RouteId, CFE_SB_DestinationD_t *NewNode){
```

Comment:
I'd consider implementing these functions using a circular list structure rather than a linear structure.

This only requires an extra "head" node which does not store data. An "empty" list only has this head node.  In this state both prev and next point back to itself (not NULL).

The advantage is that in this type of data structure there is no NULL next/prev pointers, and therefore special conditions for first/last go away.  Remove/Insert ops become simple and constant time, with no conditionals.

Iterating the list means checking for arrival at the head node again (rather than NULL) which signifies the end of iteration.

IMO this type of structure keeps things simpler and more reliable, easier to read and maintain.

Reply:
Good idea, but I'd prefer if that was a new issue/PR.  The only thing the routing does with destinations is set/get the head pointer, so it's all on the SB side (vs SBR).  I mucked minimally with naming and where I couldn't stop myself, but I am trying to limit my changes in this pr.

Reply:
See #953